### ISSUE TITLE
[heft] Heft does not clean up its subprocesses when terminating "--watch" mode

### ISSUE THREAD

**Issue by octogonz** [microsoft/rushstack#2387]
## Summary

Heft spawns subprocesses, but when Heft is killed with CTRL+C, the subprocesses are not terminated on Linux and Mac.  Windows does not have this problem.

## Repro steps

1. On Linux or Mac, run this command in **tutorials/heft-node-basic-tutorial**:
```
$ heft build --watch
```

2. Press CTRL-C

3. Save a change to a source file.

  **Expected result:** Nothing special should happen.

  **Actual result:** The shell suddenly prints this output, which is coming from an invisible background process:

```
events.js:292
      throw er; // Unhandled 'error' event
      ^

Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at process.target.send (internal/child_process.js:679:16)
    at Object.write (/home/owner/code/rushstack/apps/heft/lib/utilities/subprocess/TerminalProviderManager.js:31:22)
    at /home/owner/code/rushstack/libraries/node-core-library/lib/Terminal/Terminal.js:124:22
    at Set.forEach (<anonymous>)
    at Terminal._writeSegmentsToProviders (/home/owner/code/rushstack/libraries/node-core-library/lib/Terminal/Terminal.js:103:25)
```

The problem isn't serious, but that's not obvious from the error message. Our users were worried that something bad had happened.

## Details

Running the `ps` command after Step 2 shows that Heft's subtask is still running in the background, watching for changes to source files.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@rushstack/heft` version? |0.22.6 |
| Operating system? | Mac / Linux  |
| Would you consider contributing a PR? | Yes |

CC @MickeyPhoenix

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/2388

**Comment by octogonz** [comment]
This fix was released with Heft 0.22.7
